jellyfin/jellyfin-vue
# Add webOS/Tizen support

Support for LG WebOS 5 with Jellyfin WebOS App. A play around PR with my findings. This will likely never be merged but splitted up in junks

How to test this PR?
- Requires webos 5+ or Tizen 6+
- build with `npm build-jellyfin-web`
- WebOS: Replace the jellyfin server webinterface (keep backup or with docker just mount the directory with the files)
- WebOS: Create the `manifest.json` next to index.html (See content below)
- Tizen: Build jellyfin-tizen with the build output in `dist`

### Details
- Add new build command: `npm build-jellyfin-web`
  - Add polyfills (chrome 68) just for this command
  - Adjust vite base from `/` to `/web/` to replace the jellyfin-web server interface
  - New env var `BUILD_JELLYFIN_WEB` that can be used to skip/disable the server selection handling
- fix: missing guards for `mediaSession`
- fix: base link to config.json

Issues/Questions
- Attention: Replacing your `jellyfin-web` webinterface with `jellyfin-vue` results in a missing admin interface! Switch back if you need access to it.
- Key navigation (up/down/left/right,enter,back ...) is not implemented, so you need a magic remote
- Some jellyfin clients depend on `/web/` files, they won't even connect to a server.
  - `jellyfin-webos` needs a `manifest.json`.  Needs PWA support (or a dummy file for testing).
  - Android App: Connect is broken.
  - Other clients might be also broken.  It would be interesting to know what all jellyfin client devs need from `/web/` dir. I supposed they operate all on the api or as web wrapper

- vite legacy plugin creates a wrapper around `import.meta` ([chrome 64](https://caniuse.com/mdn-javascript_operators_import_meta)), native esm ([chrome 61](https://caniuse.com/es6-module)) and dynamic import ([chrome 63](https://caniuse.com/es6-module-dynamic-import)). Which is not required, a simple babel dependency is already enough
- `jellyfin-webos`
  - WebOS 4: Chrome: 53 (no wasm)
  - WebOS 5: Chrome: 68 (just babel)
  - WebOS 6: Chrome: 79 (just babel)
  - WebOS 22: Chrome: 87 (just babel)
  - WebOS 23: Chrome: 94 (just babel)
- `jellyfin-tizen`. I have no Tizen TV, feedback required
  - Tizen 4: Chrome 56 (no wasm)
  - Tizen 5: Chrome 63 (requires vite legacy or `import.meta` polyfill)
  - Tizen 6: Chrome 76 (just babel)

Rationals
- If polyfills shouldn't be added it might be possible to add a new `/weblegacy/` dir to jellyfin server which is just used by jellyfin clients that run with older webviews.
-  Or clients should build their own version and update with every upstream change (`jellyfin-tizen` does that)

Happy hacking!

`manifest.json` dummy
```
{
    "name": "Jellyfin",
    "description": "The Free Software Media System",
    "lang": "en-US",
    "short_name": "Jellyfin",
    "start_url": "index.html",
    "theme_color": "#101010",
    "background_color": "#101010",
    "display": "standalone",
    "icons": [
        {
            "sizes": "72x72",
            "src": "touchicon72.png",
            "type": "image/png"
        },
        {
            "sizes": "114x114",
            "src": "touchicon114.png",
            "type": "image/png"
        },
        {
            "sizes": "144x144",
            "src": "touchicon144.png",
            "type": "image/png"
        },
        {
            "sizes": "512x512",
            "src": "touchicon512.png",
            "type": "image/png"
        }
    ]
}
```
